### PR TITLE
Refactoring xplat verification changes to address the unfinished comments

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/381
Regression: No  
* Last working version:   
* How are we preventing it in future:   

## Fix

Details:
1. Remove `RequiresSigningXplatAPIs ` and use `NETCoreTargetFrameworksForSigning`, `TargetFrameworksExeForSigning ` and `TargetFrameworksLibraryForSigning ` in `common.project.props`.
2. Change the position of #if for timestamp related classes and interfaces
3. Change back the new line(without space) at the end of `Dotnet.Integration.Test.csproj` (Since I rebased from dev, and dev is correct, so there is no difference shown for this csproj file)
4. Remove net5.0 TFM from `NuGet.Frameworks.Test.csproj` and `NuGet.LibraryModel.Tests.csproj `
5. Remove unused property `NETCoreTestTargetFrameworks` in `common.project.props`

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  
Validation:

### CODE REVIEW

File: test/NuGet.Core.FuncTests/Dotnet.Integration.Test/Dotnet.Integration.Test.csproj [hunk 1]
```diff
@@ -28,4 +24,4 @@
   </ItemGroup>
   <Import Project="$(BuildCommonDirectory)common.targets" />
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
-</Project>
+</Project>
```
Comment (zivkan) [inline test/NuGet.Core.FuncTests/Dotnet.Integration.Test/Dotnet.Integration.Test.csproj]: your text editor didn't honor our `.editorconfig` setting to end the file with a newline.

**Review by zivkan — APPROVED** [review]

File: build/common.project.props [hunk 2]
```diff
@@ -14,15 +14,15 @@
     <NETFXTargetFrameworkVersion>v4.7.2</NETFXTargetFrameworkVersion>
     <NETFXTargetFramework>net472</NETFXTargetFramework>
     <NETCoreTargetFramework>netcoreapp2.1</NETCoreTargetFramework>
+    <NETCoreTargetFrameworksForSigning>$(NETCoreTargetFramework);netcoreapp5.0</NETCoreTargetFrameworksForSigning>
     <IsBuildOnlyXPLATProjects>$(DotNetBuildFromSource)</IsBuildOnlyXPLATProjects>
-    <NETCoreTargetFrameworks Condition=" '$(RequiresSigningXplatAPIs)' != 'true' ">netcoreapp2.1</NETCoreTargetFrameworks>
-    <NETCoreTargetFrameworks Condition=" '$(RequiresSigningXplatAPIs)' == 'true' ">netcoreapp2.1;netcoreapp5.0</NETCoreTargetFrameworks>
     <NETCoreTestTargetFrameworks>netcoreapp3.1;netcoreapp5.0</NETCoreTestTargetFrameworks>
```
Comment (nkolev92) [inline build/common.project.props]: If you rebase, maybe you can clean up my mess from earlier today, where NetCoreTestTargetFrameworks is unused.

Comment (heng-liu) [reply build/common.project.props]: Sure. I'll clean that.

Comment (heng-liu) [reply build/common.project.props]: Rebasing doesn't remove `NetCoreTestTargetFrameworks `. So I removed it manually.

**Review by nkolev92 — APPROVED** [review]

File: build/common.project.props (hunk 2 above)
Comment (nkolev92) [reply build/common.project.props]: oh yeah, that's why I meant, rebasing and removing it :) 
Thanks!

**Pull request merged**